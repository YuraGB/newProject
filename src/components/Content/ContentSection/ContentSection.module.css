.Section {
    margin-bottom: 1em;
    font-size: 14px;
}

.Section h5 {
    margin-bottom: 0;
    font-size: 16px;
    margin-top: 18px;
}