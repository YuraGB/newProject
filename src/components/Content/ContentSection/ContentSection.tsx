/**
 * ContentSection
 *
 * @author Yurii Huriianov <yuhur1985@gmail.com
 * @copyright 2020
 */
import React, { ReactElement } from "react";

import classes from "./ContentSection.module.css";

type Props = {
  children?: ReactElement | ReactElement[] | null;
  title?: string | undefined;
};

/**
 * ContentSection
 *
 * @param children
 * @param title
 * @return {*} JSX.Element
 */
const ContentSection: React.FC<Props> = ({
  children,
  title = "",
}): ReactElement | null => {
  if (!children && !title) {
    return null;
  }
  return (
    <section className={classes.Section}>
      <h5>{title}</h5>
      <hr />
      {children}
    </section>
  );
};

export default ContentSection;
