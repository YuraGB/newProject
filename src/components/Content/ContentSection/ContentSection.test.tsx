import React from "react";
import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import ContentSection from "./ContentSection";

describe("ContentSection tests", () => {
  const SECTION_TITLE = "Title";

  test("ContentSection with title", async () => {
    render(<ContentSection title={SECTION_TITLE} />);
    const submitButton = screen.queryByText(SECTION_TITLE);
    expect(submitButton).toBeInTheDocument();
  });

  test("match snapshots about me component", () => {
    // @ts-ignore
    const tree = renderer
      .create(<ContentSection title={SECTION_TITLE} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
