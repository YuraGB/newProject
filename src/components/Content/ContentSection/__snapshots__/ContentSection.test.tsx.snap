// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSection tests match snapshots about me component 1`] = `
<section
  className="Section"
>
  <h5>
    Title
  </h5>
  <hr />
</section>
`;
